pkgs/servers/mir/default.nix

```diff
@@ -0,0 +1,111 @@
+{ lib
+, pkgs
+, stdenv
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules
+, boost
+, egl-wayland
+, libglvnd
+, glm
+, glog
+, lttng-ust
+, udev
+, glib
+, wayland
+, libxcb
+, xorg
+, libdrm
+, mesa
+, libepoxy
+, nettle
+, libxkbcommon
+, libinput
+, libxmlxx
+, libuuid
+, pcre2
+, freetype
+, libyamlcpp
+, python39Packages
+, libevdev
+, libselinux
+, libsepol
+, pcre
+, eglexternalplatform
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mir";
+  version = "2.10.0";
+
+  src = fetchFromGitHub {
+    owner = "MirServer";
+    repo = "mir";
+    rev = "v${version}";
+    sha256 = "sha256-HPbToo7lhsHyx0w4oO2wp3zZ0xpr6SlEm2oE6Rm8b3E=";
+  };
+
+  postPatch = ''
+    # Fix error broken path found in pc file ("//")
+    for f in $(find . -name '*.pc.in') ; do
+      substituteInPlace $f \
+        --replace "/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_LIBDIR@" \
+        --replace "/@CMAKE_INSTALL_INCLUDEDIR@" "@CMAKE_INSTALL_INCLUDEDIR@"
+    done
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    extra-cmake-modules
+  ];
+
+  buildInputs = [
+    boost
+    eglexternalplatform
+    egl-wayland
+    freetype
+    glib
+    glm
+    glog
+    libdrm
+    libepoxy
+    libevdev
+    libglvnd
+    libinput
+    libselinux
+    libsepol
+    libuuid
+    libxcb
+    libxkbcommon
+    libxmlxx
+    libyamlcpp
+    lttng-ust
+    mesa
+    nettle
+    pcre
+    pcre2
+    python39Packages.pillow
+    udev
+    wayland
+    xorg.libX11
```

Comment:
xorg libraries should be declared directly, without `xorg.` redirection.

Reply:
It is actually better doing it like this because inheriting from top-level breaks splicing.

Reply:
Well, why not using `pkgs` then? As far as I remember, splicing is problematic only in very special cases, and it is fixed by explicit decoration like `splicedPackages`, not by a magic choice of "should I use xorg.libxcb or libxcb?".